lib/screens/settings/quick_links.dart

```diff
@@ -75,67 +69,81 @@ class _QuickLinksScreenState extends State<QuickLinksScreen> {
         Padding(
           padding: const EdgeInsets.only(bottom: 16),
           child: Form(
-            key: _formKey,
-            child: Row(children: [
-              PopupMenuButton<String>(
-                icon: Icon(quickLinksIconsMap[_activeIcon] ?? Icons.link),
-                onSelected: (result) {
-                  setState(() {
-                    _activeIcon = result;
-                  });
-                },
-                itemBuilder: (context) {
-                  final entries = quickLinksIconsMap.entries.toList()
-                    ..sort((a, b) => a.key.compareTo(b.key));
-                  return entries
-                      .map((entry) => PopupMenuItem(
-                          value: entry.key, child: Icon(entry.value)))
-                      .toList();
-                },
-              ),
-              Expanded(
-                child: TextFormField(
-                    controller: _textEditingController,
-                    decoration: InputDecoration(
-                        hintText: "URL",
-                        suffixIcon: IconButton(
-                            icon: const Icon(Icons.qr_code_scanner),
-                            onPressed: () {
-                              showModalBottomSheet<void>(
-                                  context: context,
-                                  builder: (context) {
-                                    return MobileScanner(
-                                        allowDuplicates: false,
-                                        onDetect: (barcode, args) {
-                                          final code = barcode.rawValue;
-                                          if (code != null) {
-                                            _textEditingController.text = code;
-                                          }
-                                          Navigator.of(context).pop();
-                                        });
-                                  });
-                            })),
-                    validator: (value) {
-                      if (value == null ||
-                          value.isEmpty ||
-                          Uri.tryParse(value) == null) {
-                        return "This doesn't look like a valid URL.";
-                      }
-                      final quickLinksModel =
-                          Provider.of<QuickLinksModel>(context, listen: false);
-                      if (quickLinksModel.sources.any(
-                          (s) => s.url.toString() == Uri.encodeFull(value))) {
-                        return "This link already exists";
-                      }
-                      return null;
+              key: _formKey,
+              child: Column(children: [
+                TextFormField(
```

Comment:
This is kinda subtle but can we hide the label field if the url is empty? The user should enter the URL first anyways so this would keep it clean when they're not adding quick links.